instrumentation/kafka/kafka-clients/kafka-clients-common/library/README.md

```diff
@@ -0,0 +1,233 @@
+# Kafka Metrics
+
+The Kafka client exposes metrics via `org.apache.kafka.common.metrics.MetricsReporter` interface.
+OpenTelemetry provides an implementation that bridges the metrics into OpenTelemetry.
+
+To use, configure `OpenTelemetryKafkaMetrics` with an OpenTelemetry instance
+via `OpenTelemetryKafkaMetrics#setOpenTelemetry(OpenTelemetry)`, and include a reference to this
+class in kafka producer or consumer configuration, i.e.:
+
+```java
+Map<String, Object> config = new HashMap<>();
+config.put(ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG, OpenTelemetryKafkaMetrics.class.getName());
+config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getKafkaConnectString());
+...
+try (KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(config)) { ... }
+```
+
+Note: Kafka reports several metrics at multiple attribute granularities. For
+example, `records-consumed-total` is reported with attribute key `[client-id]`
+and `[client-id, topic]`. If you analyze the sum of records consumed, ignoring dimensions, backends
+are likely to double count. To alleviate this, `OpenTelemetryKafkaMetrics` detects this
+scenario and only records the most granular set of attributes available. In the case
+of `records-consumed-total`, it reports `[client-id, topic]` and ignores `[client-id]`.
```

Comment:
I just added a commit with new logic that does what's described in this comment. I think its a fairly important improvement.